### PR TITLE
Flakehub cache

### PR DESCRIPTION
**Pull request by grahamc**
[FlakeHub Cache](https://flakehub.com/) is a paid extension to the Magic Nix Cache. Seamlessly upgrade repositories using Magic Nix Cache by simply subscribing for FlakeHub. Currently in [Private Beta](https://determinate.systems/).

* **One identity-aware cache means just-right access every time.** FlakeHub melds speed, security, and simplicity into one powerhouse cache that scales across projects, teams, and organizations. Eliminate the performance impact of multiple caches and accelerate CI/CD workflows with unmatched efficiency.

* **Access controls, purpose built for Nix.** Evaluating Nix demands extensive privileges, from GitHub keys to cache permissions. This poses a threat to your established security boundaries and IAM architecture.

FlakeHub is IAM reimagined for Nix. Elevate Nix security with dynamic identity-based access control, leaving behind outdated, one-size-fits-all permissions and static secrets. Authenticate and authorize with IAM roles, JSON Web Tokens (JWTs), and Single Sign-on (SSO) using trusted, policy-driven identities that align perfectly with modern DevOps and cloud-native workflows.

Register for our private beta to get early access: https://determinate.systems/

image

### CODE REVIEW

**Review by edolstra — APPROVED** [review]

File: .github/workflows/release-prs.yml [hunk 1]
```diff
@@ -19,7 +20,7 @@ jobs:
     # Only intra-repo PRs are allowed to have PR artifacts uploaded
     # We only want to trigger once the upload once in the case the upload label is added, not when any label is added
     if: |
-      github.event.pull_request.head.repo.full_name == 'DeterminateSystems/magic-nix-cache'
+      github.event.pull_request.head.repo.full_name == 'DeterminateSystems/magic-nix-cache-priv'
```
[the remark this answers is not in the document]
Comment (grahamc) [reply .github/workflows/release-prs.yml]: 🎉

File: .github/workflows/release-branches.yml [hunk 2]
```diff
@@ -6,10 +6,12 @@ on:
       # NOTE: make sure any branches here are also valid directory names,
       # otherwise creating the directory and uploading to s3 will fail
       - "main"
+      - "flakehub-cache"
```
Comment (grahamc) [inline .github/workflows/release-branches.yml:9]:
```suggestion
      - "flakehub-cache"

```

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -7,86 +7,98 @@ on:
...
       - uses: actions/checkout@v3
 
-      - uses: DeterminateSystems/nix-installer-action@main
+      - uses: DeterminateSystems/nix-installer-action@mnc
 
       - uses: DeterminateSystems/magic-nix-cache-action@main
 
       - name: Build package
-        run: "nix build .# -L --fallback"
+        run: "nix build .# -L --fallback && nix-store --export $(nix-store -qR ./result) | xz -9 > magic-nix-cache.closure.xz"
```
Comment (lucperkins) [inline .github/workflows/build.yaml:21]: Slick

**Review by lucperkins — APPROVED** [review]
I don't have much context here but nothing strikes me as particularly egregious

**Pull request merged**